synedrion/src/tools/hashing.rs

```diff
@@ -155,25 +155,32 @@ where
     let backend_modulus = modulus.as_ref();
 
     let n_bits = backend_modulus.bits_vartime();
-    let n_bytes = (n_bits + 7) / 8; // ceiling division by 8
+    let n_bytes = n_bits.div_ceil(8) as usize;
 
-    // If the number of bits is not a multiple of 8,
-    // use a mask to zeroize the high bits in the gererated random bytestring,
-    // so that we don't have to reject too much.
+    // If the number of bits is not a multiple of 8, use a mask to zeroize the high bits in the
+    // gererated random bytestring, so that we don't have to reject too much.
     let mask = if n_bits & 7 != 0 {
         (1 << (n_bits & 7)) - 1
     } else {
         u8::MAX
     };
 
+    // TODO(dp): @reviewers this uses `to_le_bytes` (little-endian) but then the original code masks out bits
+    // from the *last* byte, so it'd mask out the *low* bits yeah? Changing to mask the first byte
```

Comment:
Little-endian means the last byte is the most significant one. (I know it's confusing, it doesn't mean "little in the end", but "little end is read first")

Reply:
Right, hmm, so the high bits are in the last byte then and my `first_mut` is wrong. Changing to `last_mut`, good catch!